crates/hotshot/src/traits/networking/combined_network.rs

```diff
@@ -165,20 +168,63 @@ impl<TYPES: NodeType> CombinedNetworks<TYPES> {
 
         if !primary_failed && Self::should_delay(&message) {
             let duration = *self.delay_duration.read().await;
+            let primary_failed = Arc::clone(&self.primary_down);
             let primary_fail_counter = Arc::clone(&self.primary_fail_counter);
-            self.delayed_tasks
+            let mut receiver = self
+                .delayed_tasks_channels
                 .write()
                 .await
                 .entry(message.kind.view_number().u64())
-                .or_default()
-                .push(async_spawn(async move {
-                    async_sleep(duration).await;
-                    info!("Sending on secondary after delay, message possibly has not reached recipient on primary");
-                    primary_fail_counter.fetch_add(1, Ordering::Relaxed);
-                    secondary_future.await
-                }));
+                .or_insert_with(|| {
+                    let (s, r) = broadcast(1);
+                    (s, r.deactivate())
+                })
+                .1
+                .activate_cloned();
```

Comment:
why do we deactivate the activate the receiver right away?

Reply:
I think the map can just hold the sender side of the channel and the task can hold the receiver

Reply:
The reason for that in `SystemContextHandle` is that we have to give an active clone to the sequencer but we don't know when the sequencer will ask for it.  Here it seems like the receiver will only have on listener (the spawned task) and if the receiver is closed then the channel should be closed